OpenRA.Mods.Common/Traits/Player/TechTree.cs

```diff
@@ -130,13 +130,30 @@ public Watcher(string key, string[] prerequisites, int limit, ITechTreeElement w
 
 			bool HasPrerequisites(Cache<string, List<Actor>> ownedPrerequisites)
 			{
-				return prerequisites.All(p => !(p.Replace("~", "").StartsWith("!") ^ !ownedPrerequisites.ContainsKey(p.Replace("!", "").Replace("~", ""))));
+				// PERF: Avoid LINQ.
+				foreach (var prereq in prerequisites)
+				{
+					var withoutTilde = prereq.Replace("~", "");
+					if (withoutTilde.StartsWith("!") ^ !ownedPrerequisites.ContainsKey(withoutTilde.Replace("!", "")))
```

Comment:
Could you please add `StringComparison.Ordinal` to all these `StartsWith` while you are here?

Reply:
Added as an extra commit.